source/prefab/CodeBlock.h

```diff
@@ -0,0 +1,48 @@
+#ifndef EMP_CODE_BLOCK_H
+#define EMP_CODE_BLOCK_H
+
+#include "../web/Element.h"
+#include "../web/Widget.h"
+#include "../tools/string_utils.h"
+#include "../base/errors.h"
+
+namespace emp {
+namespace prefab {
+  /// Use CodeBlock class to easily add code to a web
+  /// application that is highlighted based on the language used.
+  class CodeBlock: public web::Element {
+    private:
+    // code element to hold user's code
+      web::Element code{emp::to_string("code")};
+
+    public:
+      /// @param code_str string of the code to be placed in code block
+      /// @param lang programming language to base syntax highlighting
+      /// List of all supported languages: https://highlightjs.org/static/demo/
+      /// Note: Due to JavaScript callback functions, can only set code for block
+      /// in constructor.
+      CodeBlock(const std::string code_str, const std::string lang, const std::string & in_name="")

```

Comment:
I’d just call it `code` not codestr

Reply:
There's a code element as a private member of the class that it referred to by `code`. Do you think I should change the element's name so the string can be called `code` instead?